x/pairing/types/types.go

```diff
@@ -39,6 +43,18 @@ const (
 // Frozen provider block const
 const FROZEN_BLOCK = math.MaxInt64
 
+func Freeze(stakeEntry *epochstoragetypes.StakeEntry) {
```

Comment:
make it a function of stake entry instead of a function that receives it as an argument